internal/ingress/controller/controller.go

```diff
@@ -687,6 +696,18 @@ func (n *NGINXController) createUpstreams(data []*extensions.Ingress, du *ingres
 					upstreams[name].LoadBalancing = anns.LoadBalancing
 				}
 
+				if upstreams[name].SessionAffinity.AffinityType == "" {
+					upstreams[name].SessionAffinity.AffinityType = anns.SessionAffinity.Type
+				}
+
+				if upstreams[name].SessionAffinity.CookieSessionAffinity.Name == "" {
+					upstreams[name].SessionAffinity.CookieSessionAffinity.Name = anns.SessionAffinity.Cookie.Name
+				}
+
+				if upstreams[name].SessionAffinity.CookieSessionAffinity.Hash == "" {
+					upstreams[name].SessionAffinity.CookieSessionAffinity.Hash = anns.SessionAffinity.Cookie.Hash
+				}
+
```

Comment:
I'd have expected no controller changes, what are these for?

Reply:
> createUpstreams creates the NGINX upstreams for each service referenced in ingress rules

If `upstreams[name].SessionAffinity.AffinityType.*` are not set here then they aren't available in nginx and therefore in the lua code.